### PR TITLE
[zIndex]  Add support for setting zIndex on markers

### PR DESCRIPTION
**Pull request by mlanter**
Add support for setting the z-index on markers via the style.

Example usage:
```
<MapView.Marker
  style={styles.importantMarker}
  ... 
/>
```

```
const styles = StyleSheet.create({
  importantMarker: {
    zIndex: 1,
  },
});
```

Test plan: 
* Verify on both iOS and Android that with this fix any markers with a zIndex > 0 show up above any other markers.
* Verify without the change sometimes markers appear below others.

### CODE REVIEW

**Comment by spikebrehm** [comment]
Awesome, thanks! Can you please add an example to the example app demonstrating this feature?  We basically use the example app as manual integration tests.

**Comment by carleryd** [comment]
Great job, just tested it and it works great! I have been wanting this feature for a long time, looking forward to seeing it in the next release!

**Comment by mlanter** [comment]
Here's an example-

screenshot-2016-10-12_13 17 34 140

**Comment by spikebrehm** [comment]
Can you also add this to Google Maps for iOS?

https://github.com/opendoor-labs/react-native-maps/blob/ff85d1f591683703be700c6bd942b665d9c8dc57/ios/AirGoogleMaps/AIRGoogleMapMarkerManager.m

**Comment by mlanter** [comment]
Yep!  Here's iOS screenshots for Apple and Google Maps-

simulator screen shot oct 13 2016 1 35 19 am
simulator screen shot oct 13 2016 1 35 26 am

File: example/examples/LoadingMap.js [hunk 1]
```diff
@@ -41,8 +41,8 @@ class LoadingMap extends React.Component {
           initialRegion={this.state.region}
           onPress={this.onMapPress}
           loadingEnabled
-          loadingIndicatorColor={"#666666"}
-          loadingBackgroundColor={"#eeeeee"}
+          loadingIndicatorColor="#666666"
+          loadingBackgroundColor="#eeeeee"
```
Comment (spikebrehm) [inline example/examples/LoadingMap.js]: 👍

**Pull request merged**